node/src/utils/round_robin.rs

```diff
@@ -143,6 +144,42 @@ where
     }
 }
 
+impl<I, K> WeightedRoundRobin<I, K>
+where
+    I: Debug,
+    K: Copy + Clone + Eq + Hash + IntoEnumIterator + Debug,
+{
+    /// Dump the contents of the queues (`Debug` representation) to a given file.
+    pub async fn debug_dump(&self, file: &mut std::fs::File) -> Result<(), std::io::Error> {
+        let mut locks = Vec::new();
+        for kind in K::into_enum_iter() {
+            let queue_guard = self
+                .queues
+                .get(&kind)
+                .expect("missing queue while dumping")
+                .queue
+                .lock()
+                .await;
+
+            locks.push((kind, queue_guard));
+        }
+
+        for (kind, guard) in locks {
+            let queue = &*guard;
+            file.write_all(format!("Queue: {:?} ({}) [\n", kind, queue.len()).as_bytes())?;
+            for (i, event) in queue.iter().enumerate() {
+                file.write_all(format!("\t{:?}\n", event).as_bytes())?;
+                if i % 1024 == 0 {
+                    file.flush()?;
```

Comment:
It might be better to wrap `file` in a `BufWriter` which should mean you can omit the explicit `flush()` calls other than the last one on line 177.

Reply:
Done.